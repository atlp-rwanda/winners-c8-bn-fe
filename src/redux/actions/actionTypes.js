export const FECTH_REQUESTS = 'FECTH_REQUESTS';
export const FECTH_REQUESTS_COMMENTS = 'FECTH_REQUESTS_COMMENTS';
export const FECTH_REQUESTS_COMMENTS_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const POST_REQUESTS_COMMENT_FAILED = 'FECTH_REQUESTS_COMMENTS_FAILED';
export const DELETE_REQUESTS_COMMENT = 'DELETE_REQUESTS_COMMENT';
export const DELETE_REQUESTS_COMMENT_FAILED = 'DELETE_REQUESTS_COMMENT_FAILED';

export const GOOGLE_SIGN_IN_START = 'GOOGLE_SIGN_IN_START';
export const GOOGLE_SIGN_IN_SUCCESS = 'GOOGLE_SIGN_IN_SUCCESS';
export const GOOGLE_SIGN_IN_FAIL = 'GOOGLE_SIGN_IN_FAIL';

export const FACEBOOK_SIGN_IN_START = 'FACEBOOK_SIGN_IN_START';
export const FACEBOOK_SIGN_IN_SUCCESS = 'FACEBOOK_SIGN_IN_SUCCESS';
export const FACEBOOK_SIGN_IN_FAIL = 'FACEBOOK_SIGN_IN_FAIL';

export const FETCH_ACCOMMODATIONS_LOADING = 'FETCH_ACCOMMODATIONS_LOADING';
export const FETCH_ACCOMMODATIONS_SUCCESS = 'FETCH_ACCOMMODATIONS_SUCCESS';
export const FETCH_ACCOMMODATIONS_FAILED = 'FETCH_ACCOMMODATIONS_FAILED';

export const FETCH_SINGLE_ACCOMMODATION_LOADING =
  'FETCH_SINGLE_ACCOMMODATION_LOADING';
export const FETCH_SINGLE_ACCOMMODATION_SUCCESS =
  'FETCH_SINGLE_ACCOMMODATION_SUCCESS';
export const FETCH_SINGLE_ACCOMMODATION_FAILED =
  'FETCH_SINGLE_ACCOMMODATION_FAILED';

export const FETCH_USER_LOADING = 'FETCH_USER_LOADING';
export const FETCH_USER_SUCCESS = 'FETCH_USER_SUCCESS';
export const FETCH_USER_FAILED = 'FETCH_USER_FAILED';

export const FETCH_ROLE_LOADING = 'FETCH_ROLE_LOADING';
export const FETCH_ROLE_SUCCESS = 'FECTH_ROLE_SUCCESS';
export const FETCH_ROLE_FAILED = 'FECTH_ROLE_FAILED';

export const ASSIGN_ROLE_LOADING = 'ASSIGN_ROLE_LOADING';
export const ASSIGN_ROLE_SUCCESS = 'ASSIGN_ROLE_SUCCESS';
export const ASSIGN_ROLE_FAILED = 'ASSIGN_ROLE_FAILED';

export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
export const LOGOUT_FAILED = 'LOGOUT_FAILED';

export const ACCOMMODATION_UPDATE_LOADING = 'ACCOMMODATION_UPDATE_LOADING';
export const ACCOMMODATION_UPDATE_SUCCESS = 'ACCOMMODATION_UPDATE_SUCCESS';
export const ACCOMMODATION_UPDATE_FAILURE = 'ACCOMMODATION_UPDATE_FAILURE';

export const ACCOMMODATION_DELETE_LOADING = 'ACCOMMODATION_DELETE_LOADING';
export const ACCOMMODATION_DELETE_SUCCESS = 'ACCOMMODATION_DELETE_SUCCESS';
export const ACCOMMODATION_DELETE_FAILURE = 'ACCOMMODATION_DELETE_FAILURE';

export const ACCOMMODATION_VIEW_LOADING = 'ACCOMMODATION_VIEW_LOADING';
export const ACCOMMODATION_VIEW_SUCCESS = 'ACCOMMODATION_VIEW_SUCCESS';
export const ACCOMMODATION_VIEW_FAILURE = 'ACCOMMODATION_VIEW_FAILURE';
export const APPROVEREJECT_PENDING_REQUEST = 'APPROVE_PENDING_REQUEST';
export const APPROVEREJECT_SUCCESS = 'APPROVE_REQUEST_SUCCESS';
export const APPROVEREJECT_ERROR = 'APPROVE_REQUEST_ERROR';

export const FETCH_ALL_CHATS = 'GET_CHATS';
export const FETCH_ALL_CHATS_FAILED = 'FAILED_TO_LOAD_CHATS';
export const CHAT_SUCCESS = 'CHAT_SUCCESS';
export const CHATTING_STATUS = 'CHATTING_STATUS';
