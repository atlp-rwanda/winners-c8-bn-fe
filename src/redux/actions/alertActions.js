
export const successAlert = (message) => {
    return {
        type: 'success-alert',
        payload: message
    }
    
};


export const failureAlert = (message) => {
    return {
        type: 'failure-alert',
        payload: message
    }
};
