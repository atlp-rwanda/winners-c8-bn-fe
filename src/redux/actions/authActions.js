const authActions ={};
authActions.login = token => ({
    type: "LOGIN",
    token
});
  
authActions.logout = () => ({
    type: "LOGOUT"
});

export default authActions;