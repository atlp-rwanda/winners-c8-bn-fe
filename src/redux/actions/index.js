// this folder will contain redux's actions