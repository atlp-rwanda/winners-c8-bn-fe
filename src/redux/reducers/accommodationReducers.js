import {
  FETCH_ACCOMMODATIONS_LOADING,
  FETCH_ACCOMMODATIONS_SUCCESS,
  FETCH_ACCOMMODATIONS_FAILED,
  LikeDislike_ACCOMMODATION_LOADING,
  LikeDislike_ACCOMMODATION_SUCCESS,
  LikeDislike_ACCOMMODATION_FAILURE,
  ACCOMMODATION_UPDATE_LOADING,
  ACCOMMODATION_UPDATE_SUCCESS,
  ACCOMMODATION_UPDATE_FAILURE,
  ACCOMMODATION_DELETE_SUCCESS,
  ACCOMMODATION_DELETE_LOADING,
  ACCOMMODATION_DELETE_FAILURE,
} from '../actions/actionTypes';


const initialState = {
    loading:false,
    accommodations: [],
    error: ''
}

export function accommodationsReducer(state = initialState, action) {
  switch (action.type) {
    case FETCH_ACCOMMODATIONS_LOADING:
      return {
        loading: true,
      };

    case FETCH_ACCOMMODATIONS_SUCCESS:
      console.log(action.payload.data);
      return {
        ...state,
        loading: false,
        accommodations: action.payload.data,
      };

    case FETCH_ACCOMMODATIONS_FAILED:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };
    case LikeDislike_ACCOMMODATION_SUCCESS:
      console.log(state.accommodations);
      return {
        ...state,
        
          accommodations: state.accommodations.map(accommodations => accommodations.id === action.payload.accommodationId? { ...accommodations, isLiked :!accommodations.isLiked, likes:accommodations.isLiked ? accommodations.likes-1 : accommodations.likes+1} : accommodations),
        
        loading: false,
        message: action.payload.res,
      };
    case LikeDislike_ACCOMMODATION_FAILURE:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };

    default:
      return state;
  }
}

// UPDATING REDUCER

export const AccommodationUpdateReducer = (state = initialState, action) => {
	switch (action.type) {
		case ACCOMMODATION_UPDATE_LOADING:
			return {
				loading: true,
				isUpdated: false
			}
	
		case ACCOMMODATION_UPDATE_SUCCESS:
			return {
				loading:false,
				data: action.payload,
				isUpdated: true
			}
		
		case ACCOMMODATION_UPDATE_FAILURE:
			return{
				loading: false,
				error: action.payload,
				isUpdated: false
			}

		default:
			return state;
	}
}

// DELETE REDUCER
export const deleteAccommodationReducer=(state=initialState, action)=>{
  switch (action.type) {
		case ACCOMMODATION_DELETE_LOADING:
			return {
				loading: true,
				isDeleted: false
			}
	
		case ACCOMMODATION_DELETE_SUCCESS:
			return {
				loading:false,
				isDeleted: true
			}
		
		case ACCOMMODATION_DELETE_FAILURE:
			return{
				loading: false,
				error: action.payload,
				isDeleted: false
			}

		default:
			return state;
	}
}

export const viewAccommodationReducer=(state=initialState, action)=>{
  switch (action.type) {
		case ACCOMMODATION_DELETE_LOADING:
			return {
				loading: true,
				isDeleted: false
			}
	
		case ACCOMMODATION_DELETE_SUCCESS:
			return {
				loading:false,
				isDeleted: true
			}
		
		case ACCOMMODATION_DELETE_FAILURE:
			return{
				loading: false,
				error: action.payload,
				isDeleted: false
			}

		default:
			return state;
	}
}
// export const LikeDislikeReducer = (state = initialState, action) => {
//   switch (action.type) {
//     case LikeDislike_ACCOMMODATION_LOADING:
//       return {
//         ...state,
//         loading: true,
//       };
//     case LikeDislike_ACCOMMODATION_SUCCESS:
// 		console.log(state.accommodations)
//       return {
//         ...state,
// 		accommodations:{
// 			...state.accommodations.accommodations,
// 			// accommodations: state.accommodations.map(accommodations => accommodations.id === action.payload.accommodationId? { ...accommodations, isLiked :!accommodations.isLiked, likes:accommodations.likes+1 } : accommodations)
// 		},
//         loading: false,
//         message: action.payload.res,
//       };
	 
// 		 default:
//           return state

//   }
// };