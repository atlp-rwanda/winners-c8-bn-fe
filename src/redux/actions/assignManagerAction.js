import { successToast, errorToast } from '../../helpers/generateToast';
import axiosInstance from '../../helpers/http';
import { FETCH_MANAGERS ,FETCH_MANAGERS_FAILED, ASSIGN_MANAGER ,ASSIGN_MANAGER_FAILED } from './actionTypes'


  export const fetchManagers = () => async (dispatch) => {
    return await 
    axiosInstance.get('/users/managers')
      .then((res) => {
        dispatch({
          type: FETCH_MANAGERS,
          payload: res.data,
        });
      })
      .catch((err) => {
        dispatch({
          type: FETCH_MANAGERS_FAILED,
          payload: 'Fetch Managers failed',
        });
      });
      
  };


  export const assignManager = (SelectedEmail, SelectedManager) => async (dispatch) => {
    return await 
    axiosInstance.patch('/users/assignManager',{email: SelectedEmail, managerId: SelectedManager})
      .then((res) => {
        if(res.status == 200){
          dispatch({
            type: ASSIGN_MANAGER,
            payload: res.data,
          });
          successToast(res.data.message)
          dispatch(fetchManagers());

        }else{
          errorToast("Only Admin can do this")
        }
      })
      .catch((err) => {
        dispatch({
          type: ASSIGN_MANAGER_FAILED,
          payload: 'Assigning Manager failed',
        });
      });
      
  };