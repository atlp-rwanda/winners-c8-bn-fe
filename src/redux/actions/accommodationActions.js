import axios from 'axios'
import { successToast, errorToast } from '../../helpers/generateToast';
import { successAlert, failureAlert} from './alertActions';
import { FETCH_ACCOMMODATIONS_LOADING, 
         FETCH_ACCOMMODATIONS_SUCCESS, 
         FETCH_ACCOMMODATIONS_FAILED,
         FETCH_SINGLE_ACCOMMODATION_LOADING,
         FETCH_SINGLE_ACCOMMODATION_SUCCESS,
         FETCH_SINGLE_ACCOMMODATION_FAILED} from "./actionTypes";
import {accommodationsUrl} from "../utils/apiUrls";
import {authHeader} from '../utils/dataSession';

export const listAccommodations = () => async dispatch => {
    dispatch({
        type: FETCH_ACCOMMODATIONS_LOADING
    });

  return await axios.get(accommodationsUrl, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_ACCOMMODATIONS_FAILED,
              payload: err.message,
              message: err
          });
          errorToast(err)
      })
}

export const detailsAccommodation = (accommodationId) => async dispatch => {
    dispatch({
        type: FETCH_SINGLE_ACCOMMODATION_LOADING
    });

  return await axios.get(accommodationsUrl+`${accommodationId}`, { headers: authHeader() })
      .then(res => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_SUCCESS,
              payload: res.data
          });
      })
      .catch(err => {
          dispatch({
              type: FETCH_SINGLE_ACCOMMODATION_FAILED,
              payload: "failed to fetch",
              message: err
          });
      })
}
