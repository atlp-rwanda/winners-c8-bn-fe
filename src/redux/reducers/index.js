// this folder will contain redux's reducers
import {
  accommodationsReducer,
  AccommodationUpdateReducer,
  deleteAccommodationReducer,
  viewAccommodationReducer,
} from './accommodationReducers';
import { combineReducers } from 'redux';
import authReducer from './authReducer';
import navbarReducer from './navbarReducer';
import notificationReducer from './notificationReducer';
import { registration } from './userReducers';
import { alert } from './alertReducers';
import {
  requestComments,
  postRequestCommentReducer,
  deleteRequestCommentReducer,
} from './requestReducer';
import resetReducer from './resetReducer';
import recoverReducer from './recoverReducer';
import userReducer from './SocialReducers';
import usersReducer from './UserReducer';
import { rolesReducer, assignRoleReducer } from './RoleReducer';
import requestsReducer from './requestReducer';
import {managers, assign_manager} from './assignManagerReducers'
import {
  fetchUserProfileReducer,
  updateUserProfileReducer,
  onlineStatusReducer,
} from './userProfileReducer';
import { statsReducer, chartStatsReducer } from './tripstatsReducers';
import PopularDestinationsReducer from './popularDestinationsReducer';

import { logoutReducer } from './logoutReducer';

export default combineReducers({
  auth: authReducer,
  navbar: navbarReducer,
  requestComments,
  postComment: postRequestCommentReducer,
  deleteComment: deleteRequestCommentReducer,
  userProfile: fetchUserProfileReducer,
  updateUserProfile: updateUserProfileReducer,
  registration,
  alert,
  reset: resetReducer,
  recover: recoverReducer,
  user: userReducer,
  accommodations: accommodationsReducer,
  updatingAccomodation: AccommodationUpdateReducer,
  deletingAccommodation: deleteAccommodationReducer,
  viewAccommodation: viewAccommodationReducer,
  users: usersReducer,
  roles: rolesReducer,
  assignRole: assignRoleReducer,
  logout: logoutReducer,
  tripStats: statsReducer,
  chartTripStats: chartStatsReducer,
  managers,
  assignManager: assign_manager,
  notification: notificationReducer,
  onlineStatus: onlineStatusReducer,
  PopularDestinations: PopularDestinationsReducer,
  requests: requestsReducer,
});
