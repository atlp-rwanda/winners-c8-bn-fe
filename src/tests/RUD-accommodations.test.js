/**
 * @jest-environment jsdom
 */
 import React from 'react';
 import configureStore from 'redux-mock-store';
 import thunk from 'redux-thunk';
 import axios from 'axios';
 import MockAdapter from 'axios-mock-adapter';
 import { listAccommodations, updateAccommodationRequest } from '../../src/redux/actions/accommodationActions';
import AxiosMockAdapter from 'axios-mock-adapter';
import axiosInstance from '../helpers/http';
const mockAxiosAdapter = new AxiosMockAdapter(axiosInstance);
const middleware = [thunk];
const mockStore = configureStore(middleware);
const mock = new MockAdapter(axios);
const store = mockStore({});
import { accommodationsReducer, AccommodationUpdateReducer, deleteAccommodationReducer } from '../../src/redux/reducers/accommodationReducers';
import { cleanup } from '@testing-library/react';
import Accommodation from '../components/ListAccomodations/Accomodations';
import { Provider } from 'react-redux';
import {
  BrowserRouter as Router,
  MemoryRouter as MemoryRouter,
} from 'react-router-dom';
import '@testing-library/jest-dom';
import renderer, { act } from 'react-test-renderer';
import {
    render,
    screen,
    fireEvent,
    waitFor,
  } from '@testing-library/react';
  import { 
    FETCH_ACCOMMODATIONS_LOADING, 
    FETCH_ACCOMMODATIONS_SUCCESS, 
    FETCH_ACCOMMODATIONS_FAILED,
    FETCH_SINGLE_ACCOMMODATION_FAILED,
    FETCH_SINGLE_ACCOMMODATION_LOADING,
    FETCH_SINGLE_ACCOMMODATION_SUCCESS,
    ACCOMMODATION_UPDATE_LOADING,
    ACCOMMODATION_UPDATE_SUCCESS,
    ACCOMMODATION_UPDATE_FAILURE,
    ACCOMMODATION_DELETE_SUCCESS,
    ACCOMMODATION_DELETE_LOADING,
    ACCOMMODATION_DELETE_FAILURE,
    ACCOMMODATION_VIEW_SUCCESS,
    ACCOMMODATION_VIEW_FAILURE,
    ACCOMMODATION_VIEW_LOADING
} from "../redux/actions/actionTypes";

describe('RUD accommodations Test', () => {
    it('It should list all accommodations', async () => {
      let response;
      mockAxiosAdapter
        .onGet(
          '/accommodations/',
          {},
          expect.objectContaining({
            Authorization: expect.stringMatching(/^Bearer/),
          })
        )
        .reply(200);
      response = await axiosInstance.get('/accommodations/');
      expect(response).toBeUndefined();
      localStorage.setItem('auth-token', 'Token');
      response = await axiosInstance.get('/accommodations/');
      expect(response.status).toBe(200);
    });

    it('should be able to handle the 401 error', async () => {
        mockAxiosAdapter.onGet(/tokenError/i).reply(401, {
        success: false,
        status: 401,
        data: { message: 'Access denied. Invalid token' },
      });
      const result = await axiosInstance.get('/tokenError');
      expect(result.status).toBe(401);
    });

    describe('accommodationsReducer(state, action)', () => {
        const state = {};
      
        const getAllAccommodations = {
          type: FETCH_ACCOMMODATIONS_SUCCESS,
          payload:'payload'
        };
      
        const FailureToGetAccommodations = {
          type: FETCH_ACCOMMODATIONS_FAILED,
          error: 'message',
        };
      
        const requestFetchingAccommodations = {
          type: FETCH_ACCOMMODATIONS_LOADING,
        };
      describe('listing accommodations', ()=>{
        it('it should return false for loading and all accommodations', () => {
            const updatedState = accommodationsReducer(state, getAllAccommodations);
            expect(updatedState).toEqual({ 
              loading: false
              });
          });
        
          it('should return false for loading and failure message', () => {
            const updatedState = accommodationsReducer(state, FailureToGetAccommodations);
            expect(updatedState).toEqual({ loading: false });
          });
        
          it('should return true for loading and send fetch req to server', () => {
            const updatedState = accommodationsReducer(state, requestFetchingAccommodations);
            expect(updatedState).toEqual({ loading: true });
          });
      })

      const updateAccommodation = {
        type: ACCOMMODATION_UPDATE_SUCCESS,
        data:'payload'
      };
    
      const FailureToUpdateAccommodation = {
        type: ACCOMMODATION_UPDATE_FAILURE,
        error: 'message',
      };
    
      const requestUpdatingAccommodations = {
        type: ACCOMMODATION_UPDATE_LOADING,
      };

      describe('updating accommodations', ()=>{
        it('it should update', async()=>{
          const store = mockStore({ 
            id: 23,
            name: "Sport view",
            description: "this is a good house, I swear.",
            location_id: 2,
            latitude: 15.234,
            longitude: -100,
            images_links: [
              "http://res.cloudinary.com/atlp8winners/image/upload/v1661447268/winners-c8-bn-be/production/accommodations/facilities/opxayjemz7cruygl3apy.jpg"
            ],
           });
      mockAxiosAdapter
      .onPatch(
        '/accommodations/23',
        {},
        expect.objectContaining({
          Authorization: expect.stringMatching(/^Bearer/),
        })
      )
      .reply(200);
          await store.dispatch(listAccommodations());
          const actions = store.getActions();
          expect(actions[1].type).toEqual(FETCH_ACCOMMODATIONS_FAILED);
        })

        it('it should return false for loading and isUpdated true', () => {
            const updatedState = AccommodationUpdateReducer(state, updateAccommodation);
            expect(updatedState).toEqual({ 
                loading: false,
				        isUpdated: true
              });
          });
        
          it('should return false for loading and failure message', () => {
            const updatedState = AccommodationUpdateReducer(state, FailureToUpdateAccommodation);
            expect(updatedState).toEqual({ loading: false, isUpdated: false });
          });
        
          it('should return true for loading and send fetch req to server', () => {
            const updatedState = AccommodationUpdateReducer(state, requestUpdatingAccommodations);
            expect(updatedState).toEqual({ loading: true, isUpdated: false });
          });
      })


      const deleteAccommodation = {
        type: ACCOMMODATION_DELETE_SUCCESS,
      };
    
      const FailureToDeleteAccommodation = {
        type: ACCOMMODATION_DELETE_FAILURE,
        error: 'message',
      };
    
      const requestDeletingAccommodations = {
        type: ACCOMMODATION_DELETE_LOADING,
      };

      describe('DELETING accommodations', ()=>{
        it('it should return false for loading and isUpdated true', () => {
            const updatedState = deleteAccommodationReducer(state, deleteAccommodation);
            expect(updatedState).toEqual({ 
                loading: false,
				        isDeleted: true
              });
          });
        
          it('it should return false for loading and failure message', () => {
            const updatedState = deleteAccommodationReducer(state, FailureToDeleteAccommodation);
            expect(updatedState).toEqual({ loading: false, isDeleted: false });
          });
        
          it('it should return true for loading and send fetch req to server', () => {
            const updatedState = deleteAccommodationReducer(state, requestDeletingAccommodations);
            expect(updatedState).toEqual({ loading: true, isDeleted: false });
          });
      })
      });
      

      describe('list accommodation(accommodations) actions', () => {

        it('dispatches FETCH_ACCOMMODATIONS_LOADING before', async () => {
          store.clearActions();
          await store.dispatch(listAccommodations());
         
        });
      
        it('dispatches FETCH_ACCOMMODATIONS_SUCCESS', async () => {
          store.clearActions();
          await store.dispatch(listAccommodations());
        });

        it('dispatches FETCH_ACCOMMODATIONS_FAILED', async () => {
          store.clearActions();
          await store.dispatch(listAccommodations());
        });
      });
      

  });
  axiosInstance;

  describe('Connected Accommodation Component to React-Redux', () => {
    let store;
    let component;
  
    beforeEach(() => {
      store = mockStore({
        accommodations: [],
        updatingAccomodation:{
            loading: true,
            isUpdated: false
        },
        deletingAccommodation:{
            loading: true,
            isDeleted:false
        },
        error: ''

      });
      mockAxiosAdapter
      .onGet(
        '/accommodations/',
        {},
        expect.objectContaining({
          Authorization: expect.stringMatching(/^Bearer/),
        })
      )
      .reply(200);
  
      component = renderer.create(
        <Provider store={store}>
          <Router>
            <Accommodation />
          </Router>
        </Provider>
      );
    });
  
    it('should render with given state from Redux store', () => {
      expect(component.toJSON()).toMatchSnapshot();
    });
  });