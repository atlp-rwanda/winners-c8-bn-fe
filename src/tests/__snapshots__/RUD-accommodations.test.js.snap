// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Accommodation Component to React-Redux should render with given state from Redux store 1`] = `
<div
  className="mainContent"
  data-testid="update-1"
>
  <div
    className="accommodation-wrapper"
  >
    <div
      className="accommodations"
    />
    <div />
    <div />
    <div />
  </div>
</div>
`;
